polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -735,6 +798,12 @@ struct ConvertPolygeistToLLVMPass
       patterns
           .add<LLVMOpLowering, GlobalOpTypeConversion, ReturnOpTypeConversion>(
               converter);
+
+      // TODO: This is a temporal solution. In the future, we might want to
+      // handle GPUDialect lowering by extending the GpuToLLVMConversionPass.
```

Comment:
Interesting, I would also have thought that the GPU to LLVM conversion pass would be able to generate correct LLVM IR in the presence of a GPUModuleOp nested in a "regular " ModuleOp.